examples/tree_2d_dgsem/elixir_ape_euler_co-rotating_vortex_pair.jl

```diff
@@ -0,0 +1,337 @@
+         (sponge_layer_min[4] <= x[2] <= sponge_layer_max[4])    # +y direction
+end
+
+function source_term_sponge_layer(u, x, t, equations::AcousticPerturbationEquations2D,
+                                  sponge_layer_min::NTuple{4}, sponge_layer_max::NTuple{4},
+                                  reference_values)
+  # Perturbed pressure source is -alpha^2 * (u - reference_value) where alpha in [0,1] is a damping
+  # factor depending on the position inside the sponge layer
+
+  # Calculate the damping factors for each direction (=0 if x is not in the corresponding sponge
+  # zone) and take the maximum. This ensures proper damping if x lies in a corner where the sponge
+  # zones for two directions overlap
+  alphas = ntuple(i -> calc_damping_factor(x, i, sponge_layer_min, sponge_layer_max),
+                                           Val(2*Trixi.ndims(equations)))
+  alpha_square = maximum(alphas)^2
+
+  return SVector(0.0, 0.0, -alpha_square*(u[3] - reference_values[1]), 0.0, 0.0, 0.0, 0.0, 0.0)
+end
+
+function source_term_sponge_layer(u, x, t, equations::CompressibleEulerEquations2D,
+                                  sponge_layer_min::NTuple{4}, sponge_layer_max::NTuple{4},
+                                  reference_values)
+  # Calculate the damping factors for each direction (=0 if x is not in the corresponding sponge
+  # zone) and take the maximum. This ensures proper damping if x lies in a corner where the sponge
+  # zones for two directions overlap
+  alphas = ntuple(i -> calc_damping_factor(x, i, sponge_layer_min, sponge_layer_max),
+                                           Val(2*Trixi.ndims(equations)))
+  alpha_square = maximum(alphas)^2 # TODO: rename this stuff
+
+  u_prim = cons2prim(u, equations)
+  s = SVector(-alpha_square*(u_prim[1] - reference_values[1]), zero(eltype(u)), zero(eltype(u)),
+              -alpha_square*(u_prim[4] - reference_values[2]))
+
+  return prim2cons(s, equations)
+end
+
+function calc_damping_factor(x, direction, sponge_layer_min, sponge_layer_max)
+  # Damping factor alpha grows linearly from 0 to 1 depending on how deep x lies in the sponge layer
+  # If x does not lie in the sponge layer, this returns 0
+
+  # Get the coordinate that determines how deep we are in the sponge zone
+  if direction in (1, 2)
+    pos = x[1]
+  else
+    pos = x[2]
+  end
+
+  # Determine where the sponge layer begins/ends to allow calculating the damping factor
+  if direction in (2, 4)
+    sponge_begin = sponge_layer_min[direction]
+    sponge_end = sponge_layer_max[direction]
+  else
+    sponge_begin = sponge_layer_max[direction]
+    sponge_end = sponge_layer_min[direction]
+  end
+
+  alpha = (pos - sponge_begin) / (sponge_end - sponge_begin)
+
+  # alpha lies in [0, 1] if and only if x lies in the sponge zone
+  if 0.0 <= alpha <= 1.0
+    return alpha
+  else
+    return 0.0
+  end
+end
+
+
+# Boundary condition for the flow problem
```

Comment:
Can you elaborate a little in the comment regarding what kind of BC is used here?